_source/_posts/2022-03-15-react-vite-number-converter.md

```diff
@@ -0,0 +1,350 @@
+---
+layout: blog_post
+title: "Creating a TypeScript React Application with Vite"
+author: holger-schmitz
+by: contractor
+communities: [javascript]
+description: "A tutorial that shows how to create a TypeScript-based React app using Vite."
+tags: [react, vite, javascript, typescript]
+tweets:
+- "⚡ Build lightning fast TypeScript React apps using Vite ⚡"
+- "Blast off into light speed! 🚀  Learn how to create a secure TypeScript-based React app powered by Vite!"
+image: blog/react-vite-number-converter/react-vite-social.jpg
+type: conversion
+github: https://github.com/oktadev/okta-react-vite-number-converter-example
+---
+
+Front-end applications are becoming ever bigger and more complex. It is not uncommon for a React app to have hundreds or even thousands of components. As the project size increases, build times become increasingly important. In large projects, you may have to wait up to a minute for the code to be translated and bundled into a production package run in the browser. The compile and load times for the development server are also a big issue for these projects. Changes in the code may only show up in the browser after a few seconds. These wait times have become a bottleneck that slows down development.
+
+[Vite](https://vitejs.dev/) addresses this problem by providing a development server and a lightning-fast build command. Vite, pronounced `/vit/`, is French for "quick," and this name describes the goal for this tool. The build command uses [Rollup](https://github.com/rollup/rollup) under the hood, which is preconfigured to create highly optimized code. The development server makes use of browser support for ES modules. Hot Module Replacement will instantly load any codebase changes into the browser.
+
+Vite was originally developed for Vue, but you can also create React and Svelte projects out of the box. In this tutorial, I will show you how to create a TypeScript-based React application using Vite. The application will be a simple number conversion tool that converts decimal numbers to hexadecimal and binary. I won't assume any prior knowledge apart from a familiarity with JavaScript.
+
+**Prerequisites**:
+
+- [Node 14](https://nodejs.org/)
+- [Okta CLI](https://cli.okta.com)
+
+**Table of Contents**{: .hide }
+* Table of Contents
+{:toc}
+
+
+## Using Vite to create the TypeScript React application
+
+Before you start, you should have a recent version of Node and npm installed on your system. The first step is to use the Vite command to create a new application. This can be done using the `npm init` command without installing any additional software. Open a terminal in a folder of your choice and run the following command.
```

Comment:
```suggestion
Before you start, you should have recent versions of Node and npm installed on your system. The first step is to use the Vite command to create a new application. This can be done using the `npm init` command without installing any additional software. Open a terminal in a folder of your choice and run the following command.
```
quibbly grammar edit